Yatima/ForLurkRepo/Eval.lean

```diff
@@ -0,0 +1,130 @@
+import Yatima.ForLurkRepo.DSL
+import Yatima.Compiler.Utils
+
+namespace Lurk
+open Std 
+
+inductive Value where
+  | lit : Literal → Value
+  | lam : List Name → Expr → Value
+  | raw : SExpr → Value
+  | env : List (Name × Value) → Value
+  deriving Repr, BEq, Inhabited
+
+abbrev Env := Std.RBMap Name Value compare
+
+open Std.Format Std.ToFormat in
+partial def Value.pprint (v : Value) (pretty := true) : Std.Format :=
+  match v with
+    | .lit l => format l 
+    | .lam ns e => paren $
+      group ("lambda" ++ line ++ paren (fmtNames ns)) ++ line ++ e.pprint pretty
+    | .raw s => s.pprint pretty
+    | .env e => sorry
+  where 
+    fmtNames (xs : List Name) := match xs with 
+      | [] => Format.nil
+      | [n]  => format $ fixName n pretty
+      | n::ns => format (fixName n pretty) ++ line ++ fmtNames ns
+
+instance : ToFormat Value where 
+  format := Value.pprint
+
+namespace Interpreter
+
+structure State where 
+  (env : Env)
+  deriving Inhabited
+
+abbrev Context := Unit 
+
+abbrev EvalM := ReaderT Context $ EStateM String State
+
+def EvalM.run (ctx : Context) (stt : State) (m : EvalM α) : 
+    EStateM.Result String State α := 
+  EStateM.run (ReaderT.run m ctx) stt 
+
+def evalUnaryOp (op : UnaryOp) (v : Value) : EvalM Value := match op with
+  | .car => default  
+  | .cdr => default
+  | .atom => default
+  | .emit => default
+
+def evalBinaryOp (op : BinaryOp) (v₁ v₂ : Value) : EvalM Value :=
+  match op with
+  | .cons => default
+  | .strcons => default
+  | .sum => match v₁, v₂ with
+    | .lit (.num x), .lit (.num y) => return .lit $ .num (x + y)
+    | _, _ => throw "error: not a number"
+  | .diff => match v₁, v₂ with
+    | .lit (.num x), .lit (.num y) => return .lit $ .num (x - y)
+    | _, _ => throw "error: not a number"
+  | .prod => match v₁, v₂ with
+    | .lit (.num x), .lit (.num y) => return .lit $ .num (x * y)
+    | _, _ => throw "error: not a number"
+  | .quot => match v₁, v₂ with
+    | .lit (.num x), .lit (.num y) => return .lit $ .num (x * y⁻¹)
+    | _, _ => throw "error: not a number"
+  | .eq => match v₁, v₂ with
+    | .lit (.num x), .lit (.num y) =>
+      return if x == y then .lit .t else .lit .nil
+    | _, _ => throw "error: not a number"
+  | .nEq => return if v₁ != v₂ then .lit .t else .lit .nil
```

Comment:
```suggestion
  | .nEq => return if v₁ == v₂ then .lit .t else .lit .nil
```
This is unfortunate naming; `nEq` doesn't mean not equal, it stands for structural equality